docs/pages/extensibility/CustomTasks.md

```diff
@@ -106,46 +108,184 @@ A custom task implementation needs to return a function with the following signa
+ * @param {string} parameters.options.projectNamespace
+ *      Namespace of the project currently being built
+ * @param {string} parameters.options.configuration
+ *      Custom task configuration, as defined in the project's ui5.yaml
+ * @param {string} parameters.options.taskName
+ *      Name of the custom task.
+ *      This parameter is only provided to custom task extensions
+ *      defining Specification Version 3.0 and later.
+ * @returns {Promise<undefined>}
+ *      Promise resolving once the task has finished
  */
 module.exports = async function({workspace, dependencies, taskUtil, options}) {
 	// [...]
 };
+
 ````
 
-The following code snippets shows an example how a task implementation could look like:
+### Required Dependencies
 
-### Example: lib/tasks/generateMarkdownFiles.js
+!!! info
+    This functionality has been added with UI5 CLI [`v3.0.0`](https://github.com/SAP/ui5-cli/releases/tag/v3.0.0)
 
-````javascript
-// Task implementation
-const markdownGenerator = require("./markdownGenerator");
+Custom tasks can export an optional callback function `determineRequiredDependencies` to control which dependency-resources are made available through the `dependencies`-reader that is provided to the task. By reducing the amount of required dependencies or by not requiring any, UI5 Tooling might be able to build a project faster.
 
-module.exports = async function({workspace, dependencies, taskUtil, options}) {
-	const textResources = await workspace.byGlob("**/*.txt");
-	const markdownResources = await markdownGenerator({
-		resources: textResources
+Before executing a task, UI5 Tooling will ensure that all required dependencies have been built.
+
+If this callback is not provided, UI5 Tooling will make an assumption as to whether the custom task requires access to any resources of dependencies based on the defined Specification Version of the custom task extension:
+
+* **Specification Version 3.0 and later:** If no callback is provided, UI5 Tooling assumes that no dependencies are required. In this case, the `dependencies` parameter will be omitted.
+* **Specification Versions before 3.0:** If no callback is provided, UI5 Tooling assumes that all dependencies are required.
+
+
+*For more details, see also [RFC 0012 UI5 Tooling Extension API v3](https://github.com/SAP/ui5-tooling/blob/rfc-ui5-tooling-extension-api-v3/rfcs/0012-UI5-Tooling-Extension-API-3.md#3-tasks-requiring-dependencies)*
+
+````javascript
+/**
+ * Callback function to define the list of required dependencies
+ *
+ * @param {object} parameters
+ * @param {Set} parameters.availableDependencies
+ *      Set containing the names of all direct dependencies of
+ *      the project currently being built.
+ * @param {function} parameters.getProject
+ *      Identical to TaskUtil#getProject.
+ *      Retrieves a Project-instance for a given project name.
+ * @param {function} parameters.getDependencies
+ *      Identical to TaskUtil#getDependencies.
```

Comment:
should use a url to the API documentation?

Reply:
Done